### PR TITLE
complete 4-module

### PR DESCRIPTION
**Pull request by JavohirMir**

### CODE REVIEW

File: docs/4-functional-programming/homework/src/fp/array.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
-import { Ord } from './ord';
+import { Ord } from "./ord";
```
Comment (Aravin) [inline docs/4-functional-programming/homework/src/fp/array.ts:1]: why are all the references of quotes updated?

File: docs/4-functional-programming/homework/src/fp/maybe.ts [hunk 2]
```diff
@@ -1,38 +1,44 @@
...
-  _tag: 'None',
+  _tag: "None",
 };
 
 /**
  * Create a Maybe instance form the value. If value(T) is nullable(null or undefined), returns None, otherwise it returns Some<T>
  */
-export const fromNullable = <T>(value: T) => ();
+export const fromNullable = <T>(value: T | null | undefined): Maybe<T> =>
+  value === null || value === undefined ? none : some(value);
```
Comment (Aravin) [inline docs/4-functional-programming/homework/src/fp/maybe.ts:33]: I think we can combine this statement as

from 
```
value === null || value === undefined ? none : some(value);
```

to 
```
value ? some(value) : none;
```

File: docs/4-functional-programming/homework/src/main.ts [hunk 3]
```diff
@@ -1,26 +1,108 @@
...
+  (executor: ExecutorUser): Either<string, string> => {
+    const clientsFilteredByDemand = clients.filter((client) => {
+      return isSome(client.demands)
+        ? client.demands.value.some((demand) =>
+            executor.possibilities.includes(demand)
+          )
+        : client;
+    });
+
+    const clientSordOrd =
```
Comment (Aravin) [inline docs/4-functional-programming/homework/src/main.ts:55]: Function name is not meaningful can we name it better?

File: docs/4-functional-programming/homework/src/fp/utils.ts [hunk 4]
```diff
@@ -1,29 +1,53 @@
...
+  fb: (b: B) => C,
+  fc: (c: C) => D
+): (a: A) => D;
+export function flow<A, B, C, D, E>(
+  fa: (a: A) => B,
+  fb: (b: B) => C,
+  fc: (c: C) => D,
+  fd: (d: D) => E
+): (a: A) => E;
 export function flow(...fns: Array<(...args: Array<any>) => any>) {
```
Comment (Aravin) [inline docs/4-functional-programming/homework/src/fp/utils.ts:23]: Can you try avoiding type `any` here?

File: docs/4-functional-programming/homework/src/fp/maybe.ts (hunk 2 above)
Comment (JavohirMirzaakbarov) [reply docs/4-functional-programming/homework/src/fp/maybe.ts:33]: Surely,  I updated it!

File: docs/4-functional-programming/homework/src/main.ts (hunk 3 above)
Comment (JavohirMirzaakbarov) [reply docs/4-functional-programming/homework/src/main.ts:55]: It was typo, I renamed it

File: docs/4-functional-programming/homework/src/fp/utils.ts (hunk 4 above)
Comment (JavohirMirzaakbarov) [reply docs/4-functional-programming/homework/src/fp/utils.ts:23]: Sure, I fixed it!

**Pull request merged**